src/entities/Explosion.cpp

```diff
@@ -161,13 +162,10 @@ void Explosion::notify_collision_with_enemy(Enemy& enemy, Sprite& enemy_sprite,
 void Explosion::try_attack_enemy(Enemy& enemy, Sprite& enemy_sprite) {
 
   // see if the enemy was already hurt by this explosion
-  for (Enemy* victim: victims) {
-    if (victim == &enemy) {
-      return;
-    }
+  auto it = std::find(victims.begin(), victims.end(), &enemy);
+  if (it == victims.end()) {
```

Comment:
It's true that I took the liberty to rework the logic of the algorithm to make it neater, but @minirop gave the right explanation :)